<a href="https://colab.research.google.com/github/chw8207/python-seminar/blob/main/Pandas_%EC%97%B0%EC%8A%B5_%EB%AC%B8%EC%A0%9C.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [43]:
import pandas as pd

In [67]:
sample = pd.read_csv("/content/drive/MyDrive/파이썬 세미나/세미나 내용/python-seminar/data/내국인 생활인구 - 내국인 생활인구.csv")
sample.head()

일자(stdr_de_id)  시간(tmzon_pd_se)  행정동코드(adstrd_code_se)   집계구코드(oa_cd)  \
0        20170226                4               11320515  1125061020007   
1        20180107               22               11380510  1115071010309   
2        20180317                6               11680610  1113073020101   
3        20190222                7               11305575  1106071020403   
4        20210706                6               11740660  1107073040015   

   생활인구합계(tot_lvpop_co)  남자0~9세(male_f0t9_lvpop_co)  \
0                4.3097                     11.9781   
1              338.3900                     23.2446   
2             1805.5256                      4.3720   
3              504.4956                     14.0599   
4              744.6219                      0.5662   

   남자10~14세(male_f10t14_lvpop_co)  남자15~19세(male_f15t19_lvpop_co)  \
0                          8.2257                          1.8759   
1                          3.1141                          1.9972   
2                        324.3596                         10.8350   
3                         14.3256                          3.5239   
4                          2.9824                          2.9719   

   남자20~24세(male_f20t24_lvpop_co)  남자25~29세(male_f25t29_lvpop_co)  ...  \
0                          4.2075                          6.2647  ...   
1                          2.6983                          0.1240  ...   
2                         12.1324                         20.4938  ...   
3                         19.3154                          6.8197  ...   
4                         31.0868                          7.2338  ...   

   여자40~44세(female_f40t44_lvpop_co)  여자45~49세(female_f45t49_lvpop_co)  \
0                            9.0816                           52.6711   
1                           10.0707                            9.9150   
2                           28.8794                           19.4895   
3                            0.2650                           36.5731   
4                           39.7471                            1.5230   

   여자50~54세(female_f50t54_lvpop_co)  여자55~59세(female_f55t59_lvpop_co)  \
0                           24.9152                            3.6202   
1                           14.6710                           29.2192   
2                            1.6087                           16.2525   
3                           14.3217                           35.7359   
4                           47.5514                           63.4768   

   여자60~64세(female_f60t64_lvpop_co)  여자65~69세(female_f65t69_lvpop_co)  \
0                           91.3448                           13.6818   
1                            0.1533                           36.7707   
2                            8.1640                           34.3989   
3                            7.2808                            6.9919   
4                           52.3638                           35.3496   

   여자70~74세(female_f70t74_lvpop_co)  여자75~79세(female_f75t79_lvpop_co)  \
0                           45.2880                            2.3141   
1                            0.9227                            4.6111   
2                            3.4030                            3.8434   
3                           27.4321                            1.8361   
4                           32.1275                           85.6665   

   여자80세이상(female_above_80_lvpop_co)      등록일자(ldadng_dt)  
0                             9.7668  2021-08-17 16:06:44  
1                            18.1781  2021-08-17 16:06:44  
2                             4.4772  2021-08-17 16:06:44  
3                             7.7510  2021-08-17 16:06:44  
4                            95.0249  2021-08-17 16:06:44  

[5 rows x 38 columns]

## 1. 컬럼명 바꾸기

sample에 있는 컬럼명에서 한글 컬럼명만 남겨주세요.  
예시: `일자(stdr_de_id)` -> `일자`

In [68]:
# 컬럼명 조회
sample.columns

Index(['일자(stdr_de_id)', '시간(tmzon_pd_se)', '행정동코드(adstrd_code_se)',
       '집계구코드(oa_cd)', '생활인구합계(tot_lvpop_co)', '남자0~9세(male_f0t9_lvpop_co)',
       '남자10~14세(male_f10t14_lvpop_co)', '남자15~19세(male_f15t19_lvpop_co)',
       '남자20~24세(male_f20t24_lvpop_co)', '남자25~29세(male_f25t29_lvpop_co)',
       '남자30~34세(male_f30t34_lvpop_co)', '남자35~39세(male_f35t39_lvpop_co)',
       '남자40~44세(male_f40t44_lvpop_co)', '남자45~49세(male_f45t49_lvpop_co)',
       '남자50~54세(male_f50t54_lvpop_co)', '남자55~59세(male_f55t59_lvpop_co)',
       '남자60~64세(male_f60t64_lvpop_co)', '남자65~69세(male_f65t69_lvpop_co)',
       '남자70~74세(male_f70t74_lvpop_co)', '남자75~79세(male_f75t79_lvpop_co)',
       '남자80세이상(male_above_80_lvpop_co)', '여자0~9세(female_f0t9_lvpop_co)',
       '여자10~14세(female_f10t14_lvpop_co)', '여자15~19세(female_f15t19_lvpop_co)',
       '여자20~24세(female_f20t24_lvpop_co)', '여자25~29세(female_f25t29_lvpop_co)',
       '여자30~34세(female_f30t34_lvpop_co)', '여자35~39세(female_f35t39_lvpop_co)',
       '여자40~44세(f

In [46]:
len(sample.columns)

38

In [69]:
# 한글 컬럼명만 남기기
sample_2 = sample.rename(columns={'일자(stdr_de_id)':'일자','시간(tmzon_pd_se)':'시간',
                         '행정동코드(adstrd_code_se)':'행정동코드','집계구코드(oa_cd)':'집계구코드',
                         '생활인구합계(tot_lvpop_co)':'생활인구합계',
                         '남자0~9세(male_f0t9_lvpop_co)':'남자0~9세',
                         '남자10~14세(male_f10t14_lvpop_co)':'남자10~14세',
                         '남자15~19세(male_f15t19_lvpop_co)':'남자15~19세',
                         '남자20~24세(male_f20t24_lvpop_co)':'남자20~24세',
                         '남자25~29세(male_f25t29_lvpop_co)':'남자25~29세',
                         '남자30~34세(male_f30t34_lvpop_co)':'남자30~34세',
                         '남자35~39세(male_f35t39_lvpop_co)':'남자35~39세',
                         '남자40~44세(male_f40t44_lvpop_co)':'남자40~44세',
                         '남자45~49세(male_f45t49_lvpop_co)':'남자45~49세',
                         '남자50~54세(male_f50t54_lvpop_co)':'남자50~54세',
                         '남자55~59세(male_f55t59_lvpop_co)':'남자55~59세',
                         '남자60~64세(male_f60t64_lvpop_co)':'남자60~64세',
                         '남자65~69세(male_f65t69_lvpop_co)':'남자65~69세',
                         '남자70~74세(male_f70t74_lvpop_co)':'남자70~74세',
                         '남자75~79세(male_f75t79_lvpop_co)':'남자75~79세',
                         '남자80세이상(male_above_80_lvpop_co)':'남자80세이상',
                         '여자0~9세(female_f0t9_lvpop_co)':'여자0~9세',
                         '여자10~14세(female_f10t14_lvpop_co)':'여자10~14세',
                         '여자15~19세(female_f15t19_lvpop_co)':'여자15~19세',
                         '여자20~24세(female_f20t24_lvpop_co)':'여자20~24세',
                         '여자25~29세(female_f25t29_lvpop_co)':'여자25~29세',
                         '여자30~34세(female_f30t34_lvpop_co)':'여자30~34세',
                         '여자35~39세(female_f35t39_lvpop_co)':'여자35~39세',
                         '여자40~44세(female_f40t44_lvpop_co)':'여자40~44세',
                         '여자45~49세(female_f45t49_lvpop_co)':'여자45~49세',
                         '여자50~54세(female_f50t54_lvpop_co)':'여자50~54세',
                         '여자55~59세(female_f55t59_lvpop_co)':'여자55~59세',
                         '여자60~64세(female_f60t64_lvpop_co)':'여자60~64세',
                         '여자65~69세(female_f65t69_lvpop_co)':'여자65~69세',
                         '여자70~74세(female_f70t74_lvpop_co)':'여자70~74세',
                         '여자75~79세(female_f75t79_lvpop_co)':'여자75~79세',
                         '여자80세이상(female_above_80_lvpop_co)':'여자80세이상',
                         '등록일자(ldadng_dt)':'등록일자'})

In [70]:
# 컬럼명 확인
sample_2.columns

Index(['일자', '시간', '행정동코드', '집계구코드', '생활인구합계', '남자0~9세', '남자10~14세',
       '남자15~19세', '남자20~24세', '남자25~29세', '남자30~34세', '남자35~39세', '남자40~44세',
       '남자45~49세', '남자50~54세', '남자55~59세', '남자60~64세', '남자65~69세', '남자70~74세',
       '남자75~79세', '남자80세이상', '여자0~9세', '여자10~14세', '여자15~19세', '여자20~24세',
       '여자25~29세', '여자30~34세', '여자35~39세', '여자40~44세', '여자45~49세', '여자50~54세',
       '여자55~59세', '여자60~64세', '여자65~69세', '여자70~74세', '여자75~79세', '여자80세이상',
       '등록일자'],
      dtype='object')

## 2. 인덱스 설정
`일자` 컬럼을 인덱스로 설정해주세요.

In [71]:
sample_3 = sample_2.set_index('일자')
sample_3.head()

시간     행정동코드          집계구코드     생활인구합계   남자0~9세  남자10~14세  남자15~19세  \
일자                                                                              
20170226   4  11320515  1125061020007     4.3097  11.9781    8.2257    1.8759   
20180107  22  11380510  1115071010309   338.3900  23.2446    3.1141    1.9972   
20180317   6  11680610  1113073020101  1805.5256   4.3720  324.3596   10.8350   
20190222   7  11305575  1106071020403   504.4956  14.0599   14.3256    3.5239   
20210706   6  11740660  1107073040015   744.6219   0.5662    2.9824    2.9719   

          남자20~24세  남자25~29세  남자30~34세  ...  여자40~44세  여자45~49세  여자50~54세  \
일자                                      ...                                 
20170226    4.2075    6.2647   13.4910  ...    9.0816   52.6711   24.9152   
20180107    2.6983    0.1240    8.4651  ...   10.0707    9.9150   14.6710   
20180317   12.1324   20.4938    1.2299  ...   28.8794   19.4895    1.6087   
20190222   19.3154    6.8197   27.6958  ...    0.2650   36.5731   14.3217   
20210706   31.0868    7.2338  115.6026  ...   39.7471    1.5230   47.5514   

          여자55~59세  여자60~64세  여자65~69세  여자70~74세  여자75~79세  여자80세이상  \
일자                                                                    
20170226    3.6202   91.3448   13.6818   45.2880    2.3141   9.7668   
20180107   29.2192    0.1533   36.7707    0.9227    4.6111  18.1781   
20180317   16.2525    8.1640   34.3989    3.4030    3.8434   4.4772   
20190222   35.7359    7.2808    6.9919   27.4321    1.8361   7.7510   
20210706   63.4768   52.3638   35.3496   32.1275   85.6665  95.0249   

                         등록일자  
일자                             
20170226  2021-08-17 16:06:44  
20180107  2021-08-17 16:06:44  
20180317  2021-08-17 16:06:44  
20190222  2021-08-17 16:06:44  
20210706  2021-08-17 16:06:44  

[5 rows x 37 columns]

일자 인덱스를 datetime 형식으로 변경한 뒤, 날짜 오름차 순으로 데이터를 정렬할 것입니다.  
astype() 메서드를 사용해서 sample.index를 str형으로 바꿔주세요.

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.astype.html

In [77]:
# 일자 인덱스를 str형으로 바꾸기
sample_3.index = sample_3.index.astype(str)

str로 변경된 인덱스를 pd.to_datetime 메서드를 사용하여 datetime 형식으로 바꿔주세요.

https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html

In [79]:
pd.to_datetime(sample_3.index)

DatetimeIndex(['2017-02-26', '2018-01-07', '2018-03-17', '2019-02-22',
               '2021-07-06', '2017-08-01', '2019-05-02', '2017-04-08',
               '2020-06-14', '2020-12-22',
               ...
               '2019-08-17', '2019-04-26', '2017-02-14', '2019-11-23',
               '2019-01-30', '2017-06-03', '2021-10-07', '2020-08-19',
               '2017-10-11', '2020-07-25'],
              dtype='datetime64[ns]', name='일자', length=500, freq=None)

sort_index() 메서드를 사용하여 주어진 데이터를 인덱스 날짜 오름차 순으로 정렬해주세요.

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sort_index.html

In [80]:
sample_4 = sample_3.sort_index()
sample_4

시간     행정동코드          집계구코드    생활인구합계   남자0~9세  남자10~14세  남자15~19세  \
일자                                                                             
20170103  15  11305606  1104054010701  381.3905  17.0115   92.1554    7.8037   
20170110   9  11620745  1116074030413  217.1740  34.4247    4.5680   10.3187   
20170112   8  11230545  1120073020201  183.6538  64.1677   16.6423   13.0121   
20170113  13  11620725  1120053020501  173.0124   6.0142   13.9943    5.8550   
20170114  13  11215850  1121064020010    3.2861   0.8931    0.0000    0.0000   
...       ..       ...            ...       ...      ...       ...       ...   
20211014  10  11215847  1125065020001  635.4962   1.5913    3.8215    3.8885   
20211018   1  11740660  1110055020202    0.7767   7.7059    0.4624   20.4231   
20211018   0  11545630  1103072010002  560.6100   4.8471    1.5185    3.8081   
20211019  23  11410690  1111056060805  202.2415  14.1716    8.1867    0.9139   
20211021  21  11650530  1108072030003  111.6846  18.2460    1.6217    9.0629   

          남자20~24세  남자25~29세  남자30~34세  ...  여자40~44세  여자45~49세  여자50~54세  \
일자                                      ...                                 
20170103    2.8387   11.4774    3.6876  ...    3.7787    7.7771    9.4217   
20170110   23.2219   11.2559    6.1775  ...    8.8282   19.1101    0.0000   
20170112   10.8813   74.4329    4.5769  ...    6.1735   33.1356    0.0009   
20170113    3.6267   21.3657   46.0382  ...    1.6809   19.6623   75.5874   
20170114    0.0000   25.7647   26.4582  ...   32.2075    9.8561   11.4491   
...            ...       ...       ...  ...       ...       ...       ...   
20211014   15.0706   23.7220   38.7037  ...   16.3752   16.4996   19.0171   
20211018   12.0564   35.7586    1.9489  ...   20.1888   11.0533   10.6035   
20211018   18.5241    5.2046    2.5614  ...   21.6098   18.8790   22.2646   
20211019    0.0000    5.8410   24.9870  ...   77.8626   22.2675   33.8264   
20211021    2.6627   16.1716   19.5965  ...   22.9154   27.0947   53.8172   

          여자55~59세  여자60~64세  여자65~69세  여자70~74세  여자75~79세  여자80세이상  \
일자                                                                    
20170103    9.4964   27.2473    5.3854   43.7826    3.2748   1.9996   
20170110    9.6557   19.4801    4.4714    1.0490   25.0163   3.5057   
20170112   46.9347    4.0340    6.5378    6.3262   14.4669  21.1275   
20170113    0.0000    2.1403   42.1636    4.0872    7.4913   7.2035   
20170114    2.1314   11.7624    2.7284   12.0665    5.3405   1.9900   
...            ...       ...       ...       ...       ...      ...   
20211014   23.5061    5.1680   28.0896    9.1223    5.5724   1.5657   
20211018   12.6074    5.7342   29.9342    5.2506    0.0000  19.8646   
20211018   13.1933   21.1251    4.1740    5.0240    3.5867  16.6064   
20211019   26.4627    4.8137    8.2692    1.5524   11.9980  18.4136   
20211021    1.6707    8.0280    1.7361    0.3644   10.3151  11.7348   

                         등록일자  
일자                             
20170103  2021-08-17 16:06:44  
20170110  2021-08-17 16:06:44  
20170112  2021-08-17 16:06:44  
20170113  2021-08-17 16:06:44  
20170114  2021-08-17 16:06:44  
...                       ...  
20211014  2021-08-17 16:06:44  
20211018  2021-08-17 16:06:44  
20211018  2021-08-17 16:06:44  
20211019  2021-08-17 16:06:44  
20211021  2021-08-17 16:06:44  

[500 rows x 37 columns]

## 3. 조건에 맞는 데이터 추출
집계구 코드가 `1107073040015` 또는 `1113073020101`인 행을 뽑아주세요.

## 4. apply 활용

각 행에 대해 남성 인구의 합을 구하여 `남자인구 합계` 이름을 가진 컬럼을 생성해주세요.

## 5. 시각화

남성인구 중 남자0\~9세가 차지하는 비율을 계산하여 "남자0~9세 비율" 컬럼을 만들어주세요.

시간의 흐름에 따라 전체 남자 생활인구 대비 `남자0~9세 비율`의 변화를 한 눈에 알 수 있도록  
`남자0~9세 비율`을 선 그래프로 나타내주세요.


* 한글 폰트가 깨지지 않도록 처리해주세요.
* 그래프 사이즈는 20\*10으로 해주세요.